Python/ceval.c

```diff
@@ -852,7 +852,22 @@ GETITEM(PyObject *v, Py_ssize_t i) {
 #define GO_TO_INSTRUCTION(op) goto PREDICT_ID(op)
 
 
-#define DEOPT_IF(cond, instname) if (cond) { goto miss; }
+#define DEOPT_IF(COND, INSTNAME)                                 \
+    if (COND) {                                                  \
+        /* This is only a single jump on release builds! */      \
+        STAT_INC(opcode, miss);                                  \
+        STAT_INC(INSTNAME, miss);                                \
+        /* The counter is always the first cache entry: */       \
+        if (ADAPTIVE_COUNTER_IS_ZERO(*next_instr)) {             \
```

Comment:
Wrap this in a `#ifdef Py_STATS` just be sure that the comment 5 lines up is true?

Reply:
I don't think you can put an `#ifdef` *inside* of a macro definition... or are you suggesting to define two different versions of `DEOPT_IF` based on `#ifdef Py_STATS`?

Reply:
Something like
```C
#ifdef Py_STATS
#define MISS_STATS(opcode, INSTR_NAME) \
...
#else
#define MISS_STATS(opcode, INSTR_NAME) ((void)0)
#endif

#define DEOPT_IF(COND, INSTNAME)                     \
    if (COND) {                            \
        MISS_STATS(opcode, INSTR_NAME);             \
        assert(_PyOpcode_Deopt[opcode] == INSTNAME);      \
        GO_TO_INSTRUCTION(INSTNAME);       \
    }
```